File: Groupe2/TP1/src/controle.py
print("Nombres divisibles par 4 mais pas par 6:")
for i in range(1, 1001):
    # Si i est divisible par 4 et non divisible par 6
    if i % 4 == 0 and i % 6 != 0:
        print(i, end=", ")

print("\nNombres pairs et divisibles par 8:")
for i in range(1, 1001):
    # Si i est pair et divisible par 8
    if i % 2 == 0 and i % 8 == 0:
        print(i, end=", ")

print("\nNombres divisibles par 5 ou 7 mais pas par 10:")
for i in range(1, 1001):
    # Si i est divisible par 5 ou 7 mais pas par 10
    if (i % 5 == 0 or i % 7 == 0) and i % 10 != 0:
        print(i, end=", ")

File: Groupe2/TP1/src/controle.c
#include <stdio.h>

int main() {
    // Divisibilité par 4 mais pas par 6 (utilisation de for)
    printf("Nombres divisibles par 4 mais pas par 6 (for):\n");
    for (int i = 1; i <= 1000; i++) {
        if (i % 4 == 0 && i % 6 != 0) {
            printf("%d, ", i);
        }
    }

    // Nombres pairs et divisibles par 8 (utilisation de while)
    printf("\nNombres pairs et divisibles par 8 (while):\n");
    int i = 1;
    while (i <= 1000) {
        if (i % 2 == 0 && i % 8 == 0) {
            printf("%d, ", i);
        }
        i++;
    }

    // Divisibilité par 5 ou 7 mais pas par 10 (utilisation de do...while)
    printf("\nNombres divisibles par 5 ou 7 mais pas par 10 (do...while):\n");
    i = 1;
    do {
        if ((i % 5 == 0 || i % 7 == 0) && i % 10 != 0) {
            printf("%d, ", i);
        }
        i++;
    } while (i <= 1000);

    return 0;
}
